ios/Codeforces Watcher/Application/features/contests/filters/models/FilterItem.swift

```diff
@@ -12,6 +12,7 @@ import common
 
 struct FilterItem {
     let title: String
-    let platform: Contest.Platform
+    let platform: Contest.Platform?
     let isOn: Bool
+    let onSwitchTap: (_ isOn: Bool) -> ()
 }
```

Comment:
I would move it to `FilterCell` and name it `UIModel`
Also it's better to pass image here instead of platform, because you use it in problems

Reply:
Also i would move filtersAdapter to separate from contests/problems folder. To `shared`, f.e.